### PR TITLE
docs: Rewrite the Proxies and HTTPS

### LINKED ISSUE #188: authorized_url is http, not https:
I'm calling flask-dance with make_slack_blueprint, and the URL flask-dance sends to Slack as the authorized_url is 

"http://mydomain.com/login/slack/authorized"

instead of the proper

"http**s**://mydomain.com/login/slack/authorized"

This means the call fails on my production server since I did not set the insecure HTTPS env variable there (and shouldn't)

how do I get flask dance to pass the https URL for the authorized_url? If I try to specify an absolute path as the authorized url then it gets treated as a relative path.

    blueprint_slack = make_slack_blueprint(
        client_id="sdfdsg242894452",
        client_secret="53019238021358rrgdf",
        scope=["identify",  "chat:write:bot"],
        **authorized_url='https://www.mydomain.com/login/slack/authorized',**   
        redirect_url='/slack_authorized',
)

If it is meaningful:

I'm running Flask 1.0+ 
- with Flask-talisman 0 all URl's redirect to https:// and I have HSTS set 
- with a gunicorn server, with relevant https flags set in my gunicorn config file

secure_proxy_ssl_header = ('HTTP_X_FORWARDED_PROTO', 'https')
forwarded_allow_ips = '*'
secure_scheme_headers = {'X-Forwarded-Proto': 'https'}
x_forwarded_for_header = 'X-FORWARDED-FOR'

PS: And yes, the client_id and secret above are bogus!

### PR DESCRIPTION
**Pull request by daenney**
* Explain which headers are necessary
* Explain how to properly access those headers
* Point to Flask's documentation on how to configure HTTPS proxies
  instead of maintaining our own, and potentially outdated
  instructions

Closes #188

### CODE REVIEW

File: docs/proxies.rst [hunk 1]
```diff
@@ -6,33 +6,40 @@ HTTPS traffic is computationally expensive. Many people running large-scale
...
+* ``X-Forwarded-Proto``: the protocol, http or https, that the request
+  came in with
 
-    from flask import Flask
-    from werkzeug.contrib.fixers import ProxyFix
+In 99.9% of the cases the `TLS termination proxy`_ will be configured to
+do the right thing by default and any well-behaved Flask application will
+work out of the box. However, if you're accessing the WSGI environment
+directly, you will run into trouble. Don't do this and instead use the
+functions provided by Werkzeug's :mod:`~werkzeug.wsgi` module or Flasks's
```
Comment (singingwolfboy) [inline docs/proxies.rst]: Typo: "Flask's"

File: docs/proxies.rst [hunk 2]
```diff
@@ -6,33 +6,40 @@ HTTPS traffic is computationally expensive. Many people running large-scale
...
+:attr:`~flask.request` to access things like a ``Host`` header.
 
-    app = Flask(__name__)
-    app.wsgi_app = ProxyFix(app.wsgi_app)
+What to do in order to ensure your Flask app is aware of the fact that
+it's being proxied, and might be behind a `TLS termination proxy`_ is
+described in Flask's own documentation, in the
+:ref:`Proxy Setup <flask:deploying-proxy-setups>` section.
```
Comment (singingwolfboy) [inline docs/proxies.rst]: This sentence structure is a bit awkward. I suggest something like:

> If your Flask app is behind a TLS termination proxy, and you need to make sure that Flask is aware of that, check Flask's documentation for :ref:\`how to deploy a proxy setup \<flask:deploying-proxy-setups\>\`.

**Comment by daenney** [comment]
@singingwolfboy Done!

**Comment by singingwolfboy** [comment]
Thanks!

**Pull request merged**